### DISCUSSION TITLE
.eq() returning empty data (get request is passing, but data is an empty array)

### DISCUSSION THREAD

**Discussion by MrPeterss** [supabase/supabase#12366] [Questions] [answered]
Hi,

I can't seem to figure out what I am doing wrong. This is practically copied/pasted from the documentation because I have almost given up at this point. I'm using Next.JS with typescript. I'm watching the network to track the request, and for some reason the request is passing, but no data is being returned (returning an empty array). I am 100% sure that my table contains a column titled "id", and I am sure that there is a table called "profiles" (although changing these values to random make the get request fail).

`
      const [loading, setLoading] = useState(false);
      const [profile, setProfile] = useState<any>(null); // this is the profile of the user
      const supabase = useSupabaseClient();
      const session = useSession();
      const user = useUser();
      
      useEffect(() => {
          getProfile()
      }, [session]);
      
      async function getProfile() {
          try {
              setLoading(true)
              if (!user) throw new Error('No user')
        
              let { data, error, status } = await supabase
                .from('profiles')
                .select()
                .eq('id', user.id)
        
              if (error && status !== 406) {
                throw error
              }
        
              if (data) {
                setProfile(data)
              }
          } catch (error) {
              console.log(error)
          } finally {
              setLoading(false)
          }
      }
`

**Comment by GaryAustin1** [comment, marked as answer]
Do you meet RLS on the client if you have it on?  Empty array is the result of the user not meeting RLS,  or your filter not finding a row with id = to user.id.   Do you actually have user.id?  And having that does not automatically mean you have a signed in user for meeting RLS.

**Reply by MrPeterss** [reply]
Thanks for replying, my user is signed in, I made sure of that. user.id successfully returns a uuid of the user that I can clearly see is in the table in supabase... not entirely sure what "RLS" is, but I'm also kinda a newbie, so I apologize for that.

**Reply by GaryAustin1** [reply]
https://supabase.com/docs/guides/auth/row-level-security
In the dashboard table UI you will see a green message in far right if RLS is enabled.  If it is you have to have a policy.
As far as signed in, I've worked with dozens of cases where the user id is populated but the jwt does not actually get to the authorization header for the call (usually when server side stuff is involved).  But if you don't have RLS enabled then that part does not apply yet, but at some point you will need RLS.

**Reply by MrPeterss** [reply]
Found the issue! Thank you for the guidance. I had RLS enabled, but I didn't have a policy set for the user to access the table. Thanks again!
